### ISSUE TITLE
Springboard crashes upon sending a message

### ISSUE THREAD

**Issue by airinlays** [iandwelker/smserver#65]
Device: iPhone 6S
iOS: 14.2

Springboard crashes entirely upon sending any message from the web interface.  Both iMessage and SMS messages cause a crash.

A message is typed into the text field.  Upon sending, the web interface freezes.  The phone also becomes unresponsive and eventually Springboard crashes.  The phone will go dark for a few seconds.

Sometimes the the phone boots into safe mode, sometimes the phone must be hard reset via the lock button and home button.

No other tweaks seem to cause interference.  SMServer was the first tweak installed onto the device after a new jailbreak.

**Comment by iandwelker** [comment]
What version of SMServer, libSMServer, and libMRYIPC do you have installed? 
Were you able to send texts without issue in the past?
Also, would you be able to get me a crash log?

**Comment by airinlays** [comment]
SMServer - 0.6.2
libSMServer - 0.5.3
libMRYIPC - 1.0

These were all downloaded straight from the twickd repo and there is no indication for an update in Cydia.

This is my first time using SMServer.  So, the first time I sent a message in the web interface, it caused a crash.

New behavior: Upon sending a message, the phone still responds to touch inputs.  I can press items on screen such as the settings icon within SMServer's application.  But things such as the control center, notification center, and all physical buttons do nothing.  Trying to activate NC, CC, or any physical input gets no reaction from the phone.  The volume popup doesn't appear.  After a few moments, the phone crashed into safe mode.

After installing Crashreporter from BigBoss, it appears to have crashed backboardd.  This makes sense in my head as Springboard crashed.  However, I have only 2 crash logs.  I definitely had more crashes than 2, but I'm not sure why they are not appearing.  Even more strange, there is a 3rd crash log underbackboardd, which appears to be SMServer related, but upon trying to open the log, it causes Crashreporter to crash.

I've attached the crash logs here as pastebin links:

https://pastebin.com/MBDrL4JJ
https://pastebin.com/ucN0tqQs

**Comment by iandwelker** [comment]
I would recommend updating libMRYIPC to 2.0 &mdash; although SMServer *should* work fine with 1.0, I'm more confident in its compatibility with libMRYIPC 2.0, so try that out and let me know if that fixes it.

**Comment by airinlays** [comment]
Well, it's been a journey.  I spent like an hour trying to find the 2.0 version of libMRYIPC and couldn't find it for the life of me.

The TLDR story: After resetting my device due to multiple errors, I reinstalled SMS server, still had libMRYIPC 1.0 installed by default, erased and refreshed my sources several times using iCleaner, and I eventually got the 2.0 version installed.  SMServer works perfectly now.  For some reason, libMRYIPC 1.0 kept getting installed alongside SMServer automatically and the 2.0 version was just never found nor detected for an upgrade through Cydia.

Works perfectly now!  Thanks a lot!  Love the tweak!  I'm using the iPhone 6S as a message server now, but I can't respond to SMS texts through it.  I think it's because my SIM is in another phone.  Not your fault, I did some Google research and I think that's something on Apple's server side.

**Comment by iandwelker** [comment]
I'm glad it's working now :) I'll close the issue
